ksql-engine/src/main/java/io/confluent/ksql/util/ExpressionTypeManager.java

```diff
@@ -191,23 +191,21 @@ protected Expression visitSubscriptExpression(
   protected Expression visitFunctionCall(final FunctionCall node,
                                          final ExpressionTypeContext expressionTypeContext) {
 
-    final UdfFactory udfFactory = functionRegistry.getUdfFactory(node.getName().getSuffix());
-    if (udfFactory != null) {
-      List<Schema.Type> argTypes = new ArrayList<>();
+    if (functionRegistry.isAggregate(node.getName().getSuffix())) {
+      final KsqlAggregateFunction ksqlAggregateFunction =
+          functionRegistry.getAggregate(
+              node.getName().getSuffix(), getExpressionSchema(node.getArguments().get(0)));
+      expressionTypeContext.setSchema(ksqlAggregateFunction.getReturnType());
```

Comment:
Wouldn't we have the same problem when we have UDAFs?

Reply:
We will, but at the moment the call `functionRegistry.isAggregate(node.getName().getSuffix())` will return false, so there will be no null pointer. Once we add the code to handle UDAFs we will need to update this also.